databricks_cli/stack/api.py

```diff
@@ -459,56 +472,115 @@ def _validate_config(self, stack_config):
-                raise StackError("{} doesn't exist in resource config".format(RESOURCE_SERVICE))
-            if RESOURCE_PROPERTIES not in resource:
-                raise StackError("{} doesn't exist in resource config".format(RESOURCE_PROPERTIES))
-            # Error on duplicate resource ID's
+            self._assert_field_in_resource(RESOURCE_ID, resource)
+            self._assert_field_in_resource(RESOURCE_SERVICE, resource)
+            self._assert_field_in_resource(RESOURCE_PROPERTIES, resource)
+
             resource_id = resource.get(RESOURCE_ID)
+            resource_service = resource.get(RESOURCE_SERVICE)
+            resource_properties = resource.get(RESOURCE_PROPERTIES)
+
+            # Error on duplicate resource ID's
             if resource_id in seen_resource_ids:
                 raise StackError("Duplicate resource ID '{}' found, please resolve.".format(
                     resource_id))
             seen_resource_ids.add(resource_id)
 
+            # Resource service-specific validations
+            if resource_service == JOBS_SERVICE:
+                self._assert_field_in_resource_properties(JOBS_RESOURCE_NAME, resource_properties,
+                                                          JOBS_SERVICE, resource_id)
+            elif resource_service == WORKSPACE_SERVICE:
+                self._assert_field_in_resource_properties(WORKSPACE_RESOURCE_PATH,
+                                                          resource_properties,
+                                                          WORKSPACE_SERVICE, resource_id)
+                self._assert_field_in_resource_properties(WORKSPACE_RESOURCE_SOURCE_PATH,
+                                                          resource_properties, WORKSPACE_SERVICE,
+                                                          resource_id)
+                self._assert_field_in_resource_properties(WORKSPACE_RESOURCE_OBJECT_TYPE,
+                                                          resource_properties, WORKSPACE_SERVICE,
+                                                          resource_id)
+            elif resource_service == DBFS_SERVICE:
+                self._assert_field_in_resource_properties(DBFS_RESOURCE_PATH, resource_properties,
+                                                          DBFS_SERVICE, resource_id)
+                self._assert_field_in_resource_properties(DBFS_RESOURCE_SOURCE_PATH,
+                                                          resource_properties, DBFS_SERVICE,
+                                                          resource_id)
+                self._assert_field_in_resource_properties(DBFS_RESOURCE_IS_DIR, resource_properties,
+                                                          DBFS_SERVICE, resource_id)
+            else:
+                raise StackError("Resource service '{}' not supported".format(resource_service))
+
+    def _assert_field_in_resource_properties(self, field, properties, service, resource_id):
```

Comment:
Most of the arguments are only used in the message, without which the function name and fuctionality could be really generic. How about
```
if resouce_service == JOB_SERVICE:
click.echo("Validating resource: $resource_id)
_assert_fields_in_dict([field1, field2], resource_properties)
elif ...
```